light/lightchain.go

```diff
@@ -298,7 +298,9 @@ func (lc *LightChain) GetBlockByHash(ctx context.Context, hash common.Hash) (*ty
 	number := lc.hc.GetBlockNumber(hash)
 	if number == nil {
 		header, err := GetHeaderByHash(ctx, lc.odr, hash)
-		if err != nil {
+		// Header may be nil, indicating the hash doesn't match any known blocks,
+		// which is a valid response to the ODR request.
+		if err != nil || header == nil {
 			return nil, errors.New("unknown block")
```

Comment:
I think it would be helpful to include the hash in this error string.

Reply:
I'd hesitate about lumping this in here.  If you search for "unknown block" in the codebase we return it from several places, including as https://github.com/celo-org/celo-blockchain/blob/1188111ac106a92c6c9d89c5ff3da265b404a161/consensus/istanbul/backend/engine.go#L63 .  In the case of users making API requests, they should know the hash they requested.  And as far as having it in the logs, it may be part of a bigger conversation worth having about having more information logged from API requests vs having the logs be done by the client or nginx gateway.  Regardless I wouldn't favor adding it to just this one.